src/transform/with_lengthscale.jl

```diff
@@ -0,0 +1,33 @@
+"""
+    with_lengthscale(kernel::Kernel, lengthscale::Real)
+    with_lengthscale(kernel::Kernel, lengthscales::AbstractVector{<:Real})
+
+Constructs a transformed kernel with `lengthscale`.
+If a vector `lengthscales` is passed instead, constructs an "ARD" kernel with different lengthscales for each dimension.
+
+The following two ways of constructing a squared-exponential kernel with
+a given lengthscale are equivalent:
+
+```jldoctest
+julia> ℓ = 2.5;
+
+julia> SqExponentialKernel() ∘ ScaleTransform(inv(ℓ)) == with_lengthscale(SqExponentialKernel(), ℓ)
+true
+```
+
+and for the ARD case:
+
+```jldoctest
+julia> ℓ = [0.5, 2.5];
+
+julia> SqExponentialKernel() ∘ ARDTransform(inv.(ℓ)) == with_lengthscale(SqExponentialKernel(), ℓ)
+true
+```
+
+"""
+function with_lengthscale(kernel::Kernel, lengthscale::Real)
+    return compose(kernel, ScaleTransform(inv(lengthscale)))
+end
+function with_lengthscale(kernel::Kernel, lengthscales::AbstractVector{<:Real})
+    return compose(kernel, ARDTransform(inv.(lengthscales)))
```

Comment:
The broadcasting machinery is not needed it seems:
```suggestion
    return kernel ∘ ARDTransform(map(inv, lengthscales))
```

Reply:
why would you prefer `map` instead of the broadcasting?

Reply:
The broadcasting machinery is very heavy and puts a lot of additional stress on the compiler, so in general I avoid it if there is no actual broadcasting of dimensions needed.